tests/policy_snapshot/generate.sh

```diff
@@ -0,0 +1,52 @@
+#!/bin/bash
+
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+set -e
+
+SNAPSHOTS_DIR_DEFAULT="./tests/policy_snapshot/snapshots"
+SECURITY_POLICIES_C_DEFAULT="./tls/s2n_security_policies.c"
+
+function display_usage {
+    echo "Usage: $0 <policy_path> [snapshots_dir] [s2n_security_policies]"
+    echo
+    echo "Arguments:"
+    echo "  policy_path                 Path to the policy util binary"
+    echo "  snapshots_dir               Path to the snapshots directory"
+    echo "                              (default: $SNAPSHOTS_DIR_DEFAULT)"
+    echo "  s2n_security_policies       Path to the s2n_security_policies.c file"
+    echo "                              (default: $SECURITY_POLICIES_C_DEFAULT)"
+    echo
+    exit 1
+}
+
+if [ $# -lt 1 ] || [ $# -gt 3 ] || [ "$1" == "--help" ]; then
+    display_usage
+fi
+
+POLICY_BINARY="$1"
+SNAPSHOTS_DIR=${2:-$SNAPSHOTS_DIR_DEFAULT}
+SECURITY_POLICIES_C=${3:-$SECURITY_POLICIES_C_DEFAULT}
+
+echo "Using snapshots directory: $SNAPSHOTS_DIR"
+echo "Using policy binary: $POLICY_BINARY"
+echo "Using security policy file: $SECURITY_POLICIES_C"
+
+echo "Extracting security policy names..."
+POLICIES=$(grep -o '{ .version = "[^"]*"' $SECURITY_POLICIES_C \
+    | sed 's/{ .version = "\(.*\)"/\1/' | grep -v "^null$")
+
+COUNT=$(echo "$POLICIES" | wc -l)
```

Comment:
Is there a chance that a malformed policy will result in ones getting missed?  Do you need  to compare the count of policies in the directory with what ends up in `$POLICIES` ?

Reply:
I don't think that'd help. Since we run the command once for every line in $POLICIES, I don't see a scenario where those two numbers wouldn't match. If we had a malformed policy of some, we'd end up writing the `policy` usage message to a file, which would still mean we'd write the file. Luckily in that case the command would fail, failing the script.

```
$ build/bin/policy
policy <version>
example: policy default_tls13

$ echo $?
1

$ build/bin/policy asdf
policy <version>
example: policy default_tls13

$ echo $?
1
```

The check would probably help if we had a source for the expected number of policies other than $POLICIES, but I'm not sure what other source we could use.

Reply:
Oh but at least for this initial commit, I manually compared the number of policies in the policies array to the number of files written and sanity-checked the files written.